### PR TITLE
Middleware

### PR DESCRIPTION
**Pull request by Sheerabth**

### CODE REVIEW

File: ango_user/app/core/exceptions.py [hunk 1]
```diff
@@ -1,10 +1,9 @@
-from fastapi import status
+from fastapi import status, HTTPException
 
 
-class AngoException(Exception):
+class AngoException(HTTPException):
     def __init__(self, status_code: int, message: str):
-        super(Exception, self).__init__(message)
-        self.status_code = status_code
+        super(AngoException, self).__init__(status_code, message)
```
Comment (RazCrimson) [inline ango_user/app/core/exceptions.py]: What is happening here?

Trying to call init of same class??
Should it not be `HTTPException`?

File: ango_user/app/routers/user.py [hunk 2]
```diff
@@ -32,16 +25,13 @@ def register_user(create_request: UserCreateRequest) -> UserDb:
...
+    user = user_service.get_user_by_email(user_email)
+    return user
 
-    except (JWTError, ValidationError):
-        raise AuthException(message="Invalid token detected")
 
-    user_service.delete(delete_request)
+@user_router.delete("/", dependencies=[Depends(authorize_user)])
+def delete_user(request: UserDeleteRequest):
+    user_service.delete(request)
```
Comment (RazCrimson) [inline ango_user/app/routers/user.py:36]: No checking for a authenticated UserA trying to send details of UserB in the request and causing a subsequent delete?

**Review by RazCrimson — CHANGES_REQUESTED** [review]

File: ango_user/app/routers/user.py (hunk 2 above)
Comment (Sheerabth) [reply ango_user/app/routers/user.py:36]: Check is there, the dependency authorize_user will do that

Comment (RazCrimson) [reply ango_user/app/routers/user.py:36]: Cool, my bad.

**Review by RazCrimson — APPROVED** [review]

**Pull request merged**